src/go/rdctl/pkg/snapshot/manager_test.go

```diff
@@ -31,20 +31,53 @@ func TestManager(t *testing.T) {
 		paths, _ := populateFiles(t, true)
 		manager := newTestManager(paths)
 		invalidNames := []string{
-			"test!",
-			`"test"`,
-			`'test'`,
-			"testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest" +
-				"testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest",
-		}
-		for _, c := range strings.Split("!$^&*()[]{};:?/'` \\\"", "") {
-			invalidNames = append(invalidNames, fmt.Sprintf("invalid%sname", c)) // spellcheck-ignore-line
+			"", // empty string not allowed
+			" can't start with a space",
+			`can't end with a "space" `,
+			// 251 characters is too long (and the indentation here is what our linter demands)
+			"12345678911234567892123456789312345678941234567895123456789612345678971234567898" +
+				"12345678991234567890123456789112345678921234567893123456789412345678951234567896" +
+				"12345678971234567898123456789912345678901234567891123456789212345678931234567894" +
+				"12345678951",
 		}
 		for _, invalidName := range invalidNames {
 			if err := manager.ValidateName(invalidName); err == nil {
 				t.Errorf("name %q is invalid but no error was returned", invalidName)
 			}
 		}
+		err := manager.ValidateName(invalidNames[3])
+		if err == nil {
+			t.Error("oversize name is invalid but no error was returned")
+		} else if !strings.Contains(err.Error(), "...") {
+			t.Errorf("No ellipsis in reported name")
+		}
+		longishNameEndingWithSpace := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ "
+		endOfLongishString := longishNameEndingWithSpace[len(longishNameEndingWithSpace)-nameDisplayCutoffSize:]
+		err = manager.ValidateName(longishNameEndingWithSpace)
+		if err == nil {
+			t.Error(" name ending with tab not caught")
```

Comment:
Why does it say `tab`? Isn't it a space?

Reply:
once upon a time it was a tab and as too often comments/strings didn't stay up to date